### ISSUE TITLE
[Magento 2.3] Move to Magento_MessageQueue instead of Rcason_Mq

### ISSUE THREAD

**Issue by peterjaap** [elgentos/LargeConfigProducts#28]
Documentation; https://devdocs.magento.com/guides/v2.3/extension-dev-guide/message-queues/message-queues.html

**Comment by peterjaap** [comment]
See PR #29 for testing

**Comment by rydemods** [comment]
I tried to install this for m2.3 but got the following error on compile
`Fatal error:  Declaration of Elgentos\LargeConfigProducts\Model\Consumer::process($productId) must be compatible with Magento\Framework\MessageQueue\ConsumerInterface::process($maxNumberOfMessages = NULL) in /var/www/magento/app/code/Elgentos/LargeConfigProducts/Model/Consumer.php on line 11`

**Comment by peterjaap** [comment]
@rydemods that PR was a brief & quick stab at trying to implement `Magento_MessageQueue`. I haven't tested or finished it yet. Might take a few more weeks/months for me to get to it (i.e. until one of our clients decides to move to 2.3).

**Comment by lfglopes** [comment]
hey, fyi

i've picked up on your work and moved a bit (emphasis on the bit) forward ...

https://github.com/lfglopes/LargeConfigProducts/commit/7ffd5a8c0bbb6730b7a00f8848f029982c8887b2

the consumer (or subscriber, depending on your glossary) seems to work fine but i can't publish messages yet, it says:

> Message queue topic "lcp.product.prewarm" is not configured

this was my attempt at a "brief & quick stab", for now. lol

**Comment by gaiterjones** [comment]
I have made the module 2.3.3 compatible and updated to use built it AQMP/RabbitMQ message queuing for dynamic prewarming. For my use I removed all the references to renatocason/magento2-module-mq as implementing both systems for 2.2.X users and 2.3.x would be too complex. Probably best if two versions of the module exist, one that still implements renatocason MQ and one that uses the integrated system in 2.3.x.

https://github.com/gaiterjones/LargeConfigProducts/commit/af64946e05349582e96673036a8cff668a7be065
